core/src/main/java/nl/nn/adapterframework/filesystem/ExchangeFileSystem.java

```diff
@@ -873,15 +885,16 @@ public Map<String, Object> getAdditionalAttachmentProperties(Attachment a) throw
 
 	@Override
 	public void createFolder(String folderName) throws FileSystemException {
-		ExchangeService exchangeService = getConnection();
+		ExchangeFolderReference reference = asFolderReference(folderName);
+		ExchangeService exchangeService = getConnection(reference);
 		boolean invalidateConnectionOnRelease = false;
 		try {
 			Folder folder = new Folder(exchangeService);
-			folder.setDisplayName(folderName);
-			folder.save(new FolderId(basefolderId.getUniqueId()));
+			folder.setDisplayName(reference.getFolderName());
+			folder.save(reference.getBaseFolderId());
```

Comment:
Hier raakt de getUniqueId() kwijt. Klopt dat?

Reply:
Dat klopt.

folder.save() verwacht een folderId object als argument. 

In de oude code werd er aan de basefolderid (wat tevens al een folderId is) gevraagd wat de unique identifier van de folder is en deze waarde werd weer gebruikt om een nieuw folderId te maken.


http://officedev.github.io/ews-java-api/docs/releases/api-2.0/apidocs/microsoft/exchange/webservices/data/property/complex/ServiceId.html#getUniqueId--
![image](https://user-images.githubusercontent.com/37303445/177278345-bbc6ee60-9bad-4b11-9eda-042cc345c8ae.png)

http://officedev.github.io/ews-java-api/docs/releases/api-2.0/apidocs/microsoft/exchange/webservices/data/property/complex/FolderId.html
![image](https://user-images.githubusercontent.com/37303445/177278678-dea023cd-377c-4c89-afa2-e4fd4468c864.png)